components/style/parallel.rs

```diff
@@ -32,7 +32,7 @@ use rayon;
 use smallvec::SmallVec;
 
 /// The minimum stack size for a thread in the styling pool, in kilobytes.
-pub const STYLE_THREAD_STACK_SIZE_KB: usize = 256;
+pub const STYLE_THREAD_STACK_SIZE_KB: usize = 512;
```

Comment:
Can we use #cfg(...) here to have different behavior for both Servo and Gecko that that this could be upstreamed one day?